imxrt1060-hal/src/gpio.rs

```diff
@@ -212,6 +219,84 @@ where
     }
 }
 
+/// GPIO input interrupt configurations.
+///
+/// These configurations do not take effect until
+/// GPIO input interrupts are enabled.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub enum InterruptConfiguration {
+    // TODO isn't it a bit repetitive to have "Sensitive" on all here?
+    LowLevelSensitive = 0,
+    HighLevelSensitive = 1,
+    RisingEdgeSensitive = 2,
+    FallingEdgeSensitive = 3,
+    EitherEdgeSensitive = 4,
+}
+
+impl<P> GPIO<P, Input>
+where
+    P: Pin,
+{
+    /// Enable (`true`) or disable (`false`) interrupts for this GPIO input.
+    pub fn set_interrupt_enable(&mut self, enable: bool) {
+        cortex_m::interrupt::free(|_| unsafe {
+            ral::modify_reg!(ral::gpio, self.register_block(), IMR, |imr| if enable {
+                imr | self.mask()
+            } else {
+                imr & !self.mask()
+            })
+        });
+    }
+
+    /// Indicates if interrupts are (`true`) or are not (`false`) enabled for this GPIO input.
+    pub fn is_interrupt_enabled(&self) -> bool {
+        unsafe { ral::read_reg!(ral::gpio, self.register_block(), IMR) & self.mask() != 0u32 }
+    }
+
+    /// Set the interrupt configuration for this GPIO input.
```

Comment:
If the user does two calls like this:

```rust
pin.set_interrupt_configuration(InterruptConfiguration::RisingEdge);

…

pin.set_interrupt_configuration(InterruptConfiguration::EitherEdge);
```

That means we have modified ICR1 or ICR2. Do we need to undo that modification to go back to `EitherEdge`?

Reply:
No need to undo anything. When `EitherEdge` is activated, the hardware ignores whatever is in the ICR register. Here's the snippet from the reference manual.

<img width="974" alt="image" src="https://user-images.githubusercontent.com/7933926/125177561-d938cb80-e1aa-11eb-8afc-36c3416957e6.png">

Reply:
Very good!